src/Show.js

```diff
@@ -1,27 +1,34 @@
-import React, {Component} from 'react';
+import Api from '@api-platform/api-doc-parser/lib/Api';
+import Resource from '@api-platform/api-doc-parser/lib/Resource';
 import {Show as BaseShow, SimpleShowLayout, TextField} from 'admin-on-rest';
-import fieldFactory from './fieldFactory';
+import PropTypes from 'prop-types';
+import React from 'react';
 
-export default class extends Component {
-  render() {
-    const factory = this.props.options.fieldFactory
-      ? this.props.options.fieldFactory
-      : fieldFactory;
+const Show = props => {
+  const {options: {api, fieldFactory, resource}} = props;
```

Comment:
disgusting :p

Reply:
Ha ha nested object destructuring is a "today" javascript!

Reply:
No. It's not.

Reply:
So is the spread operator then.